verifying/verifying.go

```diff
@@ -75,34 +79,68 @@ func Verify(p *shared.Proof, m *shared.ProofMetadata) error {
 		return fmt.Errorf("invalid indices set size; expected %d, given: %d", expectedSize, len(p.Indices))
 	}
 
-	difficulty := shared.ProvingDifficulty(numLabels, uint64(m.K1))
+	if options.verifyFunc == nil {
+		difficulty := shared.ProvingDifficulty2(numLabels, m.B, m.K1)
+		options.logger.Debug("verifying difficulty %d", difficulty)
+		options.verifyFunc = func(val uint64) bool {
+			return val < difficulty
+		}
+	}
+
 	buf := bytes.NewBuffer(p.Indices)
 	gsReader := shared.NewGranSpecificReader(buf, bitsPerIndex)
-	indicesSet := make(map[uint64]bool, m.K2)
+	indicesSet := make(map[uint64]struct{}, m.K2)
+
+	// create the ciphers for the specific nonce
+	d := shared.CalcD(numLabels, m.B)
+	offset := p.Nonce * uint32(d)
+	nonceBlock := uint8(offset / aes.BlockSize)
+
+	// since the value can be on a boundary between two blocks, we need to create two ciphers
```

Comment:
NIT (opt): it could be predicted if the value is on a boundary or if it fits in a single AES block.

Reply:
Doing 2 instead of 1 AES calls when it is not needed doesn't impact the performance of verify in a meaningful way. We profiled this function and 99% of the time used to verify a proof is spent in scrypt.

So while it is true that we could optimize this, the gain is negligible and then I'd prefer the more general approach with out handling the special case.